src/reformat/rgb_impl.rs

```diff
@@ -450,3 +450,223 @@ pub fn yuv_to_rgb_any(
     }
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn yuv_to_rgb() {
+        fn create_420(
+            matrix_coefficients: MatrixCoefficients,
+            y: &[&[u8]],
+            u: &[&[u8]],
+            v: &[&[u8]],
+        ) -> image::Image {
+            let mut yuv = image::Image {
+                width: y[0].len() as u32,
+                height: y.len() as u32,
+                depth: 8,
+                yuv_format: PixelFormat::Yuv420,
+                matrix_coefficients,
+                ..Default::default()
+            };
+            assert!(yuv.allocate_planes(decoder::Category::Color).is_ok());
+            for plane in [Plane::Y, Plane::U, Plane::V] {
+                let samples = if plane == Plane::Y {
+                    &y
+                } else if plane == Plane::U {
+                    &u
+                } else {
+                    &v
+                };
+                assert_eq!(yuv.height(plane), samples.len());
+                for y in 0..yuv.height(plane) {
+                    assert_eq!(yuv.width(plane), samples[y].len());
+                    for x in 0..yuv.width(plane) {
+                        yuv.row_mut(plane, y as u32).unwrap()[x] = samples[y][x];
+                    }
+                }
+            }
+            yuv
+        }
+        fn assert_near(yuv: &image::Image, r: &[&[u8]], g: &[&[u8]], b: &[&[u8]]) {
+            let mut dst = rgb::Image::create_from_yuv(&yuv);
+            dst.format = rgb::Format::Rgb;
+            dst.chroma_upsampling = ChromaUpsampling::Bilinear;
+            assert!(dst.allocate().is_ok());
```

Comment:
optional: you could change the function to return `AvifResult<()>` and simply use the `?` shorthand to panic on failures instead of asserting `is_ok`.

i know i haven't been consistent with that but i think we can be going forward.

Reply:
I did not find how to use that returned `AvifResult` elegantly in the test below.